lib/elixir/lib/macro.ex

```diff
@@ -2654,6 +2654,27 @@ defmodule Macro do
     end
   end
 
+  defp dbg_ast_to_debuggable({:cond, _meta, [[do: clauses]]} = ast) do
+    modified_clauses =
+      Enum.with_index(clauses, fn {:->, _meta, [[left], right]}, index ->
+        hd(
+          quote do
+            clause_value = unquote(left) ->
+              {unquote(escape(left)), clause_value, unquote(index), unquote(right)}
+          end
+        )
+      end)
```

Comment:
Instead of `with_index` + `hd`, maybe we can do something like:

```suggestion
      clauses
      |> Enum.with_index()
      |> Enum.flat_map(fn {:->, _meta, [[left], right]}, index ->
        quote do
          clause_value = unquote(left) ->
            {unquote(escape(left)), clause_value, unquote(index), unquote(right)}
        end
      end)
```

? I think it reads a little easier than `hd`, but it's probably subjective, so up to you.

Reply:
Actually I might reverse this: it worked on my local but it fails for bootstrapping reasons: we can't use the pipe.
This would defeat the readability gain. WDYT?

Reply:
Aaargh, yeah right. Let's go with `hd` then.